### PR TITLE
added numpy frontend any matrix method

### PR DESCRIPTION
**Pull request by aimaneharrak**
This is in relation with #2738. 
I worked on the numpy frontend matrix method "any" as mentioned in #1535.
I worked on the "where" parameter in the frontend method, and i had to work on broadcasting that parameter, which is why there are some loops. I didn't comment on it.
Thanks. Waiting for your feedback !

### CODE REVIEW

**Comment by simonetgordon** [comment]
Hi there,

Thanks for your contribution! There are a few tests failing and a conflict to be resolved. The best thing to do first is to update your branch with a pull from the ivy master branch. If there are still failing tests, check the 'Details' and search for the matrix method `any` to see if it is causing any failures.

Thanks!

Simone

**Comment by aimaneharrak** [comment]
Hello,
the merge conflict was resolved, and also the failing tests linked to the matrix method any,
which were formatting related.
Thanks!

**Comment by simonetgordon** [comment]
Hi there,

I believe you are also required to write the unit test for `any`. See [here](https://lets-unify.ai/ivy/deep_dive/17_ivy_frontends_tests.html) for details 😊

**Comment by aimaneharrak** [comment]
Hi,
I changed the `any` method and added its corresponding unit test.
I had some problem with the `with_out` parameter of the `test_frontend_function` because when it is set to true, the return of the function `any` is tested to be an ivy.array, and this causes a problem because `with_out` can be true, and the `out` parameter of the function be `None`. 
And if `out` is None, the return type is not always an array.
Otherwise, the function works well and passed the tests.

Thank you. Waiting for your feedback

File: ivy/functional/frontends/numpy/utility_routines/matrix_any.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import ivy
+
+
```
Comment (simonetgordon) [inline ivy/functional/frontends/numpy/utility_routines/matrix_any.py:3]: Try adding the decorator:`@handle_out_argument` here

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_utility_routines/test_matrix_any.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    where_shape_length = draw(helpers.integers(min_value=0, max_value=len(shape)))
+    if where_shape_length != 0:
+        where_nb_dims_to_change = draw(
+            helpers.integers(
+                min_value=0,
+                max_value=where_shape_length - 1
+            )
+        )
+        where_dims_to_change = [draw(
+            helpers.integers(
```
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_utility_routines/test_matrix_any.py:43]: replace `helpers.integers` with `helpers.ints`.

**Review by simonetgordon — CHANGES_REQUESTED** [review]

**Comment by simonetgordon** [comment]
Hi there,

Check out some of the review comments for what to do next. Also note that the lint test is failing due to issues with files you've modified, so have a look at the Details of that test.

Thanks!

**Comment by aimaneharrak** [comment]
Thanks for the feedback. I made the requested changes. 
There are new fails not due to the function, but this is because i synced with unifyai:master, 
and some work is going on there.

Aïmane

**Comment by aimaneharrak** [comment]
Hello, I think the function is good enough to be merged !
Waiting for your feedback

Aïmane

**Review by simonetgordon — COMMENTED** [review]
There are some binary files changed in the PR which should not be the case. Can you investigate this? Apart from this, the changes look good.

**Comment by simonetgordon** [comment]
Hi there,

Many apologies for the delay in getting back to you. Your edits are passing all the tests, nice job. However, there are extra files in the PR which need to be removed.

Thanks!

Simone

**Comment by aimaneharrak** [comment]
Hi, 

Thanks for the feedback.

I deleted the binary files, but i have no idea why there are fails.
I guess these have to do strictly with github actions ?

Aïmane

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_utility_routines/test_matrix_any.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            st.booleans()
+        ) 
+    keepdims = draw(
+        st.booleans()
+    )
+    return x, dtypes[0], axis, keepdims, where
+
+
+@given(
+    x_dtype_axis_keepdims_where=_array_with_dtype_axis_keepdims_and_where(),
```
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_utility_routines/test_matrix_any.py:70]: The issue seems to be originating from this line. Checkout the stacktrace from here: https://github.com/unifyai/ivy/runs/8255557671?check_suite_focus=true#step:6:13256

**Comment by aimaneharrak** [comment]
Hello Simone,

Thanks for your input.
I had to change the `any` function as well as the test function as there was some changes 
to the original repository. The function looks simpler now.

Aïmane

**Review by simonetgordon — COMMENTED** [review]
Hi there, this looks good! I've just added a label so that the numpy frontend tests will run. Keep an eye on the 'Details' of this check to ensure that test_any() passes.

**Comment by aimaneharrak** [comment]
Hi,

I am working on it, but I get some errors which I think come from other functions.
I am looking for ways to avoid some casting operations which block the unit test from working for 2 of the backends.

Aïmane

**Review by simonetgordon — CHANGES_REQUESTED** [review]
Hi there, I've noticed that these functions have been placed in the wrong files. Since it's a matrix method, `test_any` should be put in `ivy/ivy_tests/test_ivy/test_frontends/test_numpy/test_matrix/test_methods.py`, to shadow `test_numpy_argmax`. Please make this and other alterations to the filepath of any() (it should go in numpy/matrix/methods.py).

**Comment by aimaneharrak** [comment]
Hi,

I made the changes you requested for. The folder `...\numpy\matrix\` had not been created when i first started writing the function.

I also fixed the test for the two backends which were failing.
My apologies for the high number of commits which came from a merge with "upstream", I had an issue doing `git rebase`

Aïmane

**Review by simonetgordon — APPROVED** [review]
Looks good!

**Comment by simonetgordon** [comment]
Closes #2738

**Pull request merged**